### PR TITLE
[exporter/tanzuobservability] Support summary metrics.

### PR DESCRIPTION
**Pull request by keep94**
**Description:**
Support summary metrics in tanzu observability exporter.

**Testing:** 
Unit tests added.

### CODE REVIEW

File: exporter/tanzuobservabilityexporter/metrics.go [hunk 1]
```diff
@@ -610,3 +610,67 @@ func (regularHistogramConsumerSpec) AsHistogram(metric pdata.Metric) histogramMe
 		HistogramDataPointSlice: aHistogram.DataPoints(),
 	}
 }
+
+type summariesConsumer struct {
```
Comment (keep94) [inline exporter/tanzuobservabilityexporter/metrics.go:614]: summariesConsumer should be summaryConsumer

File: exporter/tanzuobservabilityexporter/metrics.go [hunk 2]
```diff
@@ -610,3 +610,67 @@ func (regularHistogramConsumerSpec) AsHistogram(metric pdata.Metric) histogramMe
...
+
+func (s *summariesConsumer) Consume(metric pdata.Metric, errs *[]error) {
+	summary := metric.Summary()
+	summaryDataPoints := summary.DataPoints()
+	for i := 0; i < summaryDataPoints.Len(); i++ {
+		s.sendSummaryDataPoint(metric, summaryDataPoints.At(i), errs)
+	}
+}
+
+func (*summariesConsumer) PushInternalMetrics(errs *[]error) {
```
Comment (thepeterstone) [inline exporter/tanzuobservabilityexporter/metrics.go:639]: Is this func implementing an interface? Might be nice to document it if so

File: exporter/tanzuobservabilityexporter/metrics_test.go [hunk 3]
```diff
@@ -761,6 +761,190 @@ func TestDeltaHistogramDataPointConsumerMissingBuckets(t *testing.T) {
...
+
+	sender := &mockGaugeSender{}
+	consumer := newSummaryConsumer(sender, componenttest.NewNopTelemetrySettings())
+
+	assert.Equal(t, pdata.MetricDataTypeSummary, consumer.Type())
+
+	var errs []error
+	consumer.Consume(summaryMetric, &errs)
+
+	// Summaries don't have internal metrics, but this is here for 100% test coverage.
```
Comment (thepeterstone) [inline exporter/tanzuobservabilityexporter/metrics_test.go:792]: If there's no requirement for 100% line coverage, I'd rather avoid this. I don't think it adds much value

Comment (thepeterstone) [reply exporter/tanzuobservabilityexporter/metrics_test.go:792]: Looks like the requirement is (soon to be) 90%: 
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.codecov.yml#L16

**Review by thepeterstone — COMMENTED** [review]
A couple style questions, otherwise looks good to me

File: exporter/tanzuobservabilityexporter/metrics.go (hunk 1 above)
Comment (keep94) [reply exporter/tanzuobservabilityexporter/metrics.go:614]: Done.

File: exporter/tanzuobservabilityexporter/metrics.go (hunk 2 above)
Comment (keep94) [reply exporter/tanzuobservabilityexporter/metrics.go:639]: Done.

File: exporter/tanzuobservabilityexporter/metrics_test.go (hunk 3 above)
Comment (keep94) [reply exporter/tanzuobservabilityexporter/metrics_test.go:792]: Done. Line removed.

**Review by jpkrohling — APPROVED** [review]
Not sure about the metrics parts of this, but the code itself looks sane.

@dmitryax, are you able to validate the metrics aspects of this?

**Review by dmitryax — APPROVED** [review]
LGTM

**Comment by dmitryax** [comment]
@keep94 please rebase

**Comment by keep94** [comment]
@dmitryax I just now rebased this PR

**Pull request merged**